bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/configurable/providers/ConfigUtils.java

```diff
@@ -77,24 +80,34 @@ public static Object getFiniteBalValue(String strValue, BFiniteType finiteType,
 
     public static boolean containsUnsupportedMembers(BUnionType unionType) {
         for (Type memberType : unionType.getMemberTypes()) {
-            if (!isPrimitiveOrSequenceType(TypeUtils.getImpliedType(memberType).getTag())) {
+            if (!isPrimitiveOrSequenceOrFiniteType(TypeUtils.getImpliedType(memberType).getTag())) {
                 return true;
             }
         }
         return false;
     }
 
-    private static boolean isPrimitiveOrSequenceType(int tag) {
-        return tag == TypeTags.NULL_TAG || tag <= TypeTags.BOOLEAN_TAG || TypeTags.isXMLTypeTag(tag);
+    private static boolean isPrimitiveOrSequenceOrFiniteType(int tag) {
+        return tag == TypeTags.NULL_TAG || tag <= TypeTags.BOOLEAN_TAG || TypeTags.isXMLTypeTag(tag) ||
+                tag == TypeTags.FINITE_TYPE_TAG;
     }
 
     public static Object getUnionValue(VariableKey key, BUnionType unionType, String value, String arg) {
         List<Object> matchingValues = getConvertibleMemberValues(value, unionType);
+        List<Object> assignableValues = new ArrayList<>();
         if (matchingValues.size() == 1) {
-            return matchingValues.get(0);
+            return matchingValues.getFirst();
+        }
+        for (int i = 0; i < unionType.getMemberTypes().size() && i < matchingValues.size(); i++) {
```

Comment:
`i < unionType.getMemberTypes().size() && i < matchingValues.size()` Why we do this kind of check?

Reply:
```ballerina
enum Fruit {
    APPLE,
    ORANGE,
    BANANA
}

type FiniteColor string|"Red" | "Green";
type Foo FiniteColor|"Blue";
// configurable Fruit? fruit = ();
configurable Foo? finiteValueColor = ();
public function main() {
    Foo? color = finiteValueColor;
}
```
At line 96, we should simply return the values that match the union type. If more than one value matches, then we should throw an error. In that case, this additional check is unnecessary.